### PR TITLE
Endpoint to edit a future event by admin

### LINKED ISSUE #136: API: Create an api to edit an event that is not past
Create an API endpoint that enables edition on a specific event that is not past. 

The endpoint should take as parameters the event_id and the user_id and consider that only Admin role can edit events. It should filter the date of the event so it makes sure that it's only available for future events.

This API will be used on this component - #134 

The endpoint has to be documented on Swagger

### PR DESCRIPTION
**Pull request by AnilMula**
# Description

Endpoint to edit a future event by admin

Fixes #136 

# How to test?

It can be tested with postman, make sure to change event_date to a future date

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/server/api/controllers/events.controller.js [hunk 1]
```diff
@@ -68,10 +68,33 @@ const deleteEvent = async (userId, eventId) => {
...
+const editEvent = async (eventId, userId, updatedEvent) => {
+  const users = [];
+  await knex('users')
+    .select('users.id')
+    .join('user_roles', 'users.id', '=', 'user_roles.user_id')
+    .where('user_roles.role_id', 3)
+    .then((records) =>
+      records.forEach((record) => {
+        users.push(record.id);
+      }),
```
Comment (tpetrina) [inline src/server/api/controllers/events.controller.js:80]: This is imperative code. While it is working, you should know how to rewrite it in functional style with `map`.

File: src/server/api/routes/events.router.js [hunk 2]
```diff
@@ -137,4 +137,56 @@ router.delete('/:userId/:eventId', (req, res) => {
...
+ *            title:
+ *              type: string
+ *            startDate:
+ *              type: string
+ *              format: date-time
+ *            endDate:
+ *              type: string
+ *              format: date-time
+ *            classId:
+ *              type: string
```
Comment (tpetrina) [inline src/server/api/routes/events.router.js:169]: Are you allowing editing of all these properties? They should match the last part of the functionality in the other file.

File: src/server/api/routes/events.router.js [hunk 3]
```diff
@@ -137,4 +137,56 @@ router.delete('/:userId/:eventId', (req, res) => {
...
+ */
+
+router.patch('/:eid/:uid', (req, res) => {
+  eventsController
+    .editEvent(req.params.eid, req.params.uid, req.body)
+    .then((result) => {
+      if (result === 0) {
+        res.status(404).send('The event ID you provided does not exist.');
+      } else if (result === 'not an admin') {
+        res.status(404).send(`${req.params.uid} is not an administrator`);
```
Comment (tpetrina) [inline src/server/api/routes/events.router.js:184]: If you are not an administrator, return 403 which means forbidden.

File: src/server/api/controllers/events.controller.js [hunk 4]
```diff
@@ -68,10 +68,33 @@ const deleteEvent = async (userId, eventId) => {
...
+    );
+  if (users.includes(parseInt(userId, 10))) {
+    const today = moment().format('YYYY-MM-DD HH:mm:ss');
+    return knex('events')
+      .where({ id: eventId })
+      .where('event_date', '>', today)
+      .update({
+        venue: updatedEvent.venue,
```
Comment (tpetrina) [inline src/server/api/controllers/events.controller.js:88]: Do you want to allow anything else to be edited besides venue?

**Review by tpetrina — CHANGES_REQUESTED** [review]
Good stuff, but I'd like to see more properties editable. Also, Swagger should be fixed a bit.

File: src/server/api/routes/events.router.js (hunk 2 above)
Comment (AnilMula) [reply src/server/api/routes/events.router.js:169]: only venue editable, removing all unnecessary

File: src/server/api/routes/events.router.js (hunk 3 above)
Comment (AnilMula) [reply src/server/api/routes/events.router.js:184]: changed

File: src/server/api/controllers/events.controller.js (hunk 4 above)
Comment (AnilMula) [reply src/server/api/controllers/events.controller.js:88]: I didn't get this , are there any missing column names? like updated date?

File: src/server/api/controllers/events.controller.js (hunk 1 above)
Comment (AnilMula) [reply src/server/api/controllers/events.controller.js:80]: used map

File: src/server/api/controllers/events.controller.js (hunk 4 above)
Comment (tpetrina) [reply src/server/api/controllers/events.controller.js:88]: You could edit max participants, description, venue date and event type.

File: src/server/api/controllers/events.controller.js [hunk 5]
```diff
@@ -86,6 +86,10 @@ const editEvent = async (eventId, userId, updatedEvent) => {
       .where('event_date', '>', today)
       .update({
         venue: updatedEvent.venue,
+        max_participants: updatedEvent.max_participants,
+        description: updatedEvent.description,
+        event_date: updatedEvent.event_date,
+        event_type: updatedEvent.event_type,
```
Comment (tpetrina) [inline src/server/api/controllers/events.controller.js:92]: Properties on the right side should be `maxParticipants` , not `max_participants`.

**Review by tpetrina — CHANGES_REQUESTED** [review]
Almost there!

File: src/server/api/controllers/events.controller.js (hunk 5 above)
Comment (AnilMula) [reply src/server/api/controllers/events.controller.js]: changed to camel case for all these three names and also in swagger

**Review by tpetrina — APPROVED** [review]

**Review by Yaser-Aldammad — APPROVED** [review]

**Pull request merged**